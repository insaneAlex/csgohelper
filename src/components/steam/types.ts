import {NumObject} from '@/types';

export enum PriceOptions {
  DAY = '24_hours',
  WEEK = '7_days',
  MONTH = '30_days',
  ALL = 'all_time'
}

export type GridConfigType = Record<string, NumObject>;
