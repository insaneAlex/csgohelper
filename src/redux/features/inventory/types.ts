import {SteamIDType} from '../../types';

export enum InventoryStatuses {
  IDLE = 'idle',
  INIT_LOAD = 'Init loading',
  FORCE_LOAD = 'Force loading',
  NO_PROFILE = 'Request failed with status 404',
  PRIVATE_INVENTORY = 'Request failed with status 403'
}

export type GetInventoryPayloadType = SteamIDType & {isForceUpdate?: boolean};
