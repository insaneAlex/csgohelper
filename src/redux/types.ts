import {InventoryItemType} from '@/types';
import {CS2InventoryFetchErrorType} from '@/pages/api/csgoInventory';

export type SteamIDType = {steamid: string};
export type InventoryErrorType = null | string | CS2InventoryFetchErrorType;

export type InventoryState = {
  isLoading: boolean;
  error: InventoryErrorType;
  items: InventoryItemType[];
  update_time?: string | null;
};
