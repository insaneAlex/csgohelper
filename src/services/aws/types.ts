import {SteamProfileType} from '@/core/types';
import {InventoryItemType} from '..';
import {PriceType} from '../types';

export type AWSConfigType = {region: string; credentials: {accessKeyId: string; secretAccessKey: string}};

export type AmazonResponseType = {
  '$metadata': {
    httpStatusCode: number;
    requestId: string;
    extendedRequestId?: string;
    cfId?: string;
    attempts: number;
    totalRetryDelay: number;
  };
};

export type PricesType = Record<string, {price: PriceType}> | null;

export type NoPriceInventory = Omit<InventoryItemType, 'prices'>[];
export type InventoryRecordType = {inventory?: NoPriceInventory; update_time?: string; profile?: SteamProfileType};
export type InventoryCacheType = Record<string, InventoryRecordType>;
