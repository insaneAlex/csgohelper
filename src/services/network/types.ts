type InternalFetchOptions = {
  params?: Record<string | number, string | string[] | number | boolean | undefined>;
} & Omit<RequestInit, 'body' | 'credentials'>;

export type UrlParam = string | number | boolean | string[] | null | undefined;

export type FetchOptions = Omit<InternalFetchOptions, 'method'>;

export type RequestBody = Record<string | number, string | number | boolean | undefined | object | null>;
