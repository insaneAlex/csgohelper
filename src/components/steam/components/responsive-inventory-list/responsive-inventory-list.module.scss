@import 'styles/index.scss';

.wrapper {
  flex-direction: column;
  margin: 0 10px 10px 10px;
  @include box-container;

  .list {
    display: flex;
    align-content: flex-start;
    flex-wrap: wrap;
  }

  .updateTime {
    font-size: 0.9em;
    text-align: center;
    padding: 0 10px;

    @include body-short;
  }

  .overlay {
    background: rgba($base-dark, 0.8);
    z-index: $z-index-normal;

    @include portal-overlay;
  }
}
