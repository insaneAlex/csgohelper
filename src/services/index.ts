export {AWS_REGION, INVENTORY_TABLE} from './constants';
export {isClient} from './is-client';
export {storage} from './storage';
export {noop} from './noop';
export type {PriceType} from './types';
