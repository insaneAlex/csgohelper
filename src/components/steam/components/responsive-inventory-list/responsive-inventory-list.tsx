import {InventoryItemType} from '@/src/services/steam-inventory';
import {itemsUpdateTimeSelector} from '@/src/redux';
import {InventoryItem} from '../inventory-item';
import {useSelector} from 'react-redux';
import React, {FC} from 'react';

import styles from './responsive-inventory-list.module.scss';

type Props = {items: InventoryItemType[]};

export const ResponsiveInventoryList: FC<Props> = ({items}) => {
  const updateTime = useSelector(itemsUpdateTimeSelector);
  return (
    <section className={styles.wrapper}>
      {updateTime && <p className={styles.updateTime}>inventory cached, last update - {updateTime}</p>}
      <div className={styles.list}>
        {items.map((item) => (
          <InventoryItem key={item.assetid} item={item} imgSize={{width: 110, height: 82}} />
        ))}
      </div>
    </section>
  );
};
