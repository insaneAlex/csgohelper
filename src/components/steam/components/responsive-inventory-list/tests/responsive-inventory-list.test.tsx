import {InventoryItemType} from '@/src/services/steam-inventory';
import {render, screen} from '@testing-library/react';

import {PriceOptions} from '../../../types';
import {NextRouter} from 'next/router';
import {ResponsiveInventoryList} from '../responsive-inventory-list';

const dispatchMock = jest.fn();
const selectedId = 'id2';
jest.mock('@/src/services', () => ({}));
jest.mock('react-redux', () => ({useSelector: jest.fn(), useDispatch: () => dispatchMock}));
const routerMock = {useRouter: () => ({push: jest.fn()}), query: {item: selectedId}} as unknown as NextRouter;
const defaultProps = {router: routerMock};
describe('ResponsiveInventoryList', () => {
  const itemName = 'itmName';
  const item = {
    icon_url: 'icn',
    name_color: 'clr',
    name: itemName,
    assetid: 'id1',
    exterior: 'f',
    prices: {[PriceOptions.WEEK]: {average: 2.004}}
  } as unknown as InventoryItemType;
  it('should render correctly', () => {
    const {container} = render(<ResponsiveInventoryList {...defaultProps} items={[item]} />);
    expect(container).toMatchSnapshot();
  });
  describe('when some item is selected', () => {
    it('should render overlay', () => {
      render(<ResponsiveInventoryList {...defaultProps} items={[item, {...item, assetid: selectedId}]} />);
      const overlay = screen.getByTestId('overlay');
      expect(overlay).toBeInTheDocument();
    });
  });
});
