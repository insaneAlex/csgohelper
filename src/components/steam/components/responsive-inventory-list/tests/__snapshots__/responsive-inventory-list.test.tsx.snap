// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveInventoryList should render correctly 1`] = `
<div>
  <section
    class="wrapper"
  >
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <div
          class="link"
          data-isopen="false"
        >
          <div
            class="header"
          >
            <img
              alt="itmName"
              height="82"
              loading="lazy"
              src="https://steamcommunity-a.akamaihd.net/economy/image/icn"
              width="110"
            />
            <p
              class="describe"
            >
              itmName
            </p>
          </div>
          <span>
            2.00$
          </span>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
